Add Support for Nested Packages in list

ThomasThelen opened the pull request:
I created a couple of methods in dataone_register.py that contains code which was duplicated before. You'll see some removed lines with the new function call replacement in the harvester.

The changes in repository.py are documentation and a small pep8 suggestion (`params` wasn't being used).

Testing Files:
  I renamed the datone_test & associated json file to harvester_test, and then createds a new test for the dataone stuff (dataone_test.py).
The testing updates puts us at 89%-we'll see how circleci reports it on the web.

This addresses issue #61

Xarthisius commented on plugin_tests/dataone_register_test.py:
```diff
@@ -0,0 +1,320 @@
+        package = get_package_list("https://search.dataone.org/#view/urn:uuid:6f5533ab-6508-4ac7-82a3-1df88ed4580e")
+
+        # Metadata that should be returned
+        expected_result = {
```
Can you export `expected_result` to a file and read it from disk to make this more readable?

Xarthisius commented on plugin_tests/dataone_register_test.py:
```diff
@@ -0,0 +1,320 @@
+
+        # In this case, the primary metadata object is the same as the metadata object
+        fileList = get_package_files(data, full_meta_data, full_meta_data)
+        assert fileList == expected_result
```
Could you use `self.assertDictEqual` for `dict` comparisons? It yields much nicer tracebacks than raw `assert` in case something goes awry.

Xarthisius commented on server/dataone_register.py:
```diff
@@ -263,47 +285,93 @@ def get_documents(package_pid):
+    fileList = {}
+    for fileObj in data:
+        try:
+            fileName = fileObj['fileName']
```
Could you use `dict.get()` instead? `fileName = fileObj.get('fileName', fileObj['identifier'])` Is 'identifier' always present?

Xarthisius commented on server/dataone_register.py:
```diff
@@ -263,47 +285,93 @@ def get_documents(package_pid):
+        except KeyError:
+            fileName = fileObj['identifier']
+        try:
+            fileSize = fileObj['size']
```
Same here: `fileName = int(fileObj.get('size', 0))`. Possibly catch `ValueError` if 'size' is non integer.

Xarthisius commented:
I enabled test build for forks. It should run with the next commit.

ThomasThelen commented:
@Xarthisius 

I moved the nested json structure to a new file and named it to match the harvester test.

I switched dictionary access to using the get method-which allowed my to get rid of some of the try catch blocks.

I also switched the bogus urls to localhost_01 and dictionary asserts to assertDictEqual.

amoeba commented:
Thanks @ThomasThelen !
